Library/Homebrew/dependency.rb

```diff
@@ -145,3 +145,31 @@ def installed?
     false
   end
 end
+
+class PrioritizedTapDependency < Dependency
+  attr_reader :tap
+
+  def initialize(name, tags=[], env_proc=DEFAULT_ENV_PROC, option_name=name)
+    @tap, _, option_name = option_name.rpartition "/"
```

Comment:
If we don't keep the constructor the same as `Dependency` we need to create a new version of `Dependency#merge_repeats` since it constructs `Dependency` instances there.